sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/cryptography/EncryptedBlockBlobAsyncClient.java

```diff
@@ -0,0 +1,416 @@
+     * of a block blob's, use PutBlock and PutBlockList on a regular BlockBlob. For more information, see the
+     * <a href="https://docs.microsoft.com/rest/api/storageservices/put-blob">Azure Docs</a>.
+     * <p>
+     * Note that the data passed must be replayable if retries are enabled (the default). In other words, the
+     * {@code Flux} must produce the same data each time it is subscribed to.
+     * <p>
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.blob.cryptography.EncryptedBlockBlobAsyncClient.uploadWithResponse#Flux-long-BlobHTTPHeaders-Metadata-AccessTier-BlobAccessConditions}
+     *
+     * @param data The data to write to the blob. Note that this {@code Flux} must be replayable if retries are enabled
+     * (the default). In other words, the Flux must produce the same data each time it is subscribed to.
+     * @param length The exact length of the data. It is important that this value match precisely the length of the
+     * data emitted by the {@code Flux}.
+     * @param headers {@link BlobHTTPHeaders}
+     * @param metadata {@link Metadata}
+     * @param tier {@link AccessTier} for the destination blob.
+     * @param accessConditions {@link BlobAccessConditions}
+     * @return A reactive response containing the information of the uploaded block blob.
+     */
+    public Mono<Response<BlockBlobItem>> uploadWithResponse(Flux<ByteBuffer> data, long length, BlobHTTPHeaders headers,
+        Metadata metadata, AccessTier tier, BlobAccessConditions accessConditions) {
+        return withContext(context -> uploadWithResponse(data, length, headers, metadata, tier, accessConditions,
+            context));
+    }
+
+    Mono<Response<BlockBlobItem>> uploadWithResponse(Flux<ByteBuffer> data, long length, BlobHTTPHeaders headers,
+        Metadata metadata, AccessTier tier, BlobAccessConditions accessConditions, Context context) {
+        final Metadata metadataFinal = metadata == null ? new Metadata() : metadata;
+        final BlobAccessConditions accessConditionsFinal = accessConditions == null ? new BlobAccessConditions()
+            : accessConditions;
+        Mono<Flux<ByteBuffer>> dataFinal = encryptionPolicy.prepareToSendEncryptedRequest(data, metadataFinal);
+        // Readjust the length to account for padding.
+        long lengthFinal =  length + (16 - length % 16);
+
+        return dataFinal.flatMap(df ->
+            postProcessResponse(this.azureBlobStorage.blockBlobs().uploadWithRestResponseAsync(null, null, df,
+                lengthFinal, null, metadataFinal, tier, null, headers,
+                accessConditionsFinal.getLeaseAccessConditions(), null /*cpk*/,
+                accessConditionsFinal.getModifiedAccessConditions(), context))
+            .map(rb -> new SimpleResponse<>(rb, new BlockBlobItem(rb.getDeserializedHeaders()))));
```

Comment:
Ideally this would pass through to the unencrypted client, but I seem to recall some conversation about why that wasn't possible for this api in particular. Can you leave a comment as to what that reason is/was